forge/forge/agent/components.py

```diff
@@ -1,9 +1,12 @@
 from __future__ import annotations
 
 from abc import ABC
-from typing import Callable, TypeVar
+from typing import Callable, Generic, Optional, TypeVar
+
+from pydantic import BaseModel
 
 T = TypeVar("T", bound="AgentComponent")
```

Comment:
I know this isn't part of the PR; just a note that `T` being a bound TypeVar is unusual and therefore possibly confusing.

Reply:
I know you've mentioned it somewhere already and I cannot find any guidelines about it. Looking at our codebase I see initials, e.g. `SC - SystemConfiguration`. So the better name would be `AC`/`C`?

Reply:
Yes. And noted, I'll add it to the code guidelines I'm writing.